cmd/otel-allocator/Dockerfile

```diff
@@ -1,9 +1,27 @@
 # Build the otel-allocator binary
-FROM golang:1.20-alpine as builder
+FROM mcr.microsoft.com/oss/go/microsoft/golang:1.20 as builder
```

Comment:
update our doc to keep it at 1.20 until we update TA, when we update the other golang versions